app/controllers/dockers_controller.rb

```diff
@@ -0,0 +1,32 @@
+class DockersController < ApplicationController
+  action_auth_level :index, :instructor
+  def index; end
+
+  action_auth_level :uploadDockerImage, :instructor
+  def uploadDockerImage
+    image_name = params[:image_name]
+    if image_name.nil? || image_name.empty?
+      flash[:error] = "Please specify an image name."
+    elsif %r{^[a-z0-9](_{0,2}[a-z0-9])*(/[a-z0-9](_{0,2}[a-z0-9])*)?$}.match(image_name).nil?
```

Comment:
You should use `\A` and `\z` delimiters instead of `^` and `$` since those only match up to a newline. 
![Screenshot 2024-02-08 at 01 28 19](https://github.com/autolab/Autolab/assets/9074856/c8c56930-5ce9-4999-875c-cca1cd1c7051)

See https://stackoverflow.com/questions/577653/difference-between-a-z-and-in-ruby-regular-expressions for more details